openstack/cloudeyeservice/v2/alarmrule/requests.go

```diff
@@ -0,0 +1,145 @@
+package alarmrule
+
+import (
+	"log"
+
+	"github.com/chnsz/golangsdk"
+)
+
+type DimensionOpts struct {
+	Name  string `json:"name" required:"true"`
+	Value string `json:"value,omitempty"`
+}
+
+type PolicyOpts struct {
+	MetricName string `json:"metric_name" required:"true"`
+	// The value can be 0
+	Period             int    `json:"period"`
+	Filter             string `json:"filter" required:"true"`
+	ComparisonOperator string `json:"comparison_operator" required:"true"`
+	// The Value ranges from 0 to MAX_VALUE
+	Value float64 `json:"value"`
+	Unit  string  `json:"unit,omitempty"`
+	Count int     `json:"count" required:"true"`
+	// The value can be 0
+	SuppressDuration int `json:"suppress_duration"`
+	Level            int `json:"level,omitempty"`
+}
+
+type NotificationOpts struct {
+	Type             string   `json:"type" required:"true"`
+	NotificationList []string `json:"notification_list" required:"true"`
+}
+
+type CreateOpts struct {
+	Name                  string             `json:"name" required:"true"`
+	Description           string             `json:"description,omitempty"`
+	Namespace             string             `json:"namespace" required:"true"`
+	ResourceGroupID       string             `json:"resource_group_id,omitempty"`
+	Resources             [][]DimensionOpts  `json:"resources" required:"true"`
+	Policies              []PolicyOpts       `json:"policies" required:"true"`
+	Type                  string             `json:"type" required:"true"`
+	AlarmNotifications    []NotificationOpts `json:"alarm_notifications,omitempty"`
+	OkNotifications       []NotificationOpts `json:"ok_notifications,omitempty"`
+	NotificationBeginTime string             `json:"notification_begin_time,omitempty"`
+	NotificationEndTime   string             `json:"notification_end_time,omitempty"`
+	EnterpriseProjectID   string             `json:"enterprise_project_id,omitempty"`
+	Enabled               bool               `json:"enabled"`
+	NotificationEnabled   bool               `json:"notification_enabled"`
+	AlarmTemplateID       string             `json:"alarm_template_id,omitempty"`
+}
+
+func Create(c *golangsdk.ServiceClient, opts CreateOpts) (r CreateResult) {
+	b, err := golangsdk.BuildRequestBody(opts, "")
+	if err != nil {
+		r.Err = err
+		return
+	}
+	log.Printf("[DEBUG] create AlarmRule url:%q, body=%#v, opt=%#v", rootURL(c), b, opts)
```

Comment:
we don't need this log message in golangsdk

Reply:
fixed